core/src/main/java/com/datastax/oss/driver/internal/core/resolver/mockResolver/MockResolverFactory.java

```diff
@@ -0,0 +1,67 @@
+package com.datastax.oss.driver.internal.core.resolver.mockResolver;
+
+import com.datastax.oss.driver.internal.core.resolver.AbstractResolverFactory;
+import com.datastax.oss.driver.internal.core.resolver.Resolver;
+import com.datastax.oss.driver.internal.core.resolver.Response;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Produces instances of {@link MockResolver}. All {@link Resolver} instances produced by instance
+ * of this class return results predefined by the factory. Results are determined by the provided
+ * mapping from {@link String} hostnames to {@link Response}s. Starts with empty mapping and {@code
+ * null} default response.
```

Comment:
```suggestion
 * Produces instances of {@link MockResolver}. All {@link Resolver} instances produced by instance
 * of this class return results sourced by the factory.
 * if {@link defaultResponse} is not set, it produces same results as `DefaultResolver` for hosts that are not in the {@link responses} 
```

Reply:
>  if {@link defaultResponse} is not set, it produces same results as `DefaultResolver` for hosts that are not in the {@link responses} 

This is not currently true. If it's not set it will return null. Should I change the logic? Since DefaultResolver does not do anything smart I can create a new instance inside and reroute the call to it

Reply:
Nevermind, I think it is ok. The factory returns null but they are handled inside MockResolver. I got lost a bit.